### PR TITLE
view history order

### PR DESCRIPTION
**Pull request by ghost**
order and cancel order

## Related Tickets
- ticket redmine
- https://edu-redmine.sun-asterisk.vn/issues/73188

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image

image
image
Detail: 
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controllers/Order.controller.ts [hunk 1]
```diff
@@ -215,3 +221,104 @@ export const postOrder = async (
     next(error);
   }
 };
+
+function getStatusText(statusNumber: number): string {
```
Comment (huongnt-2545) [inline src/controllers/Order.controller.ts]: cái này e dùng enum đc nhé

File: src/controllers/Order.controller.ts [hunk 2]
```diff
@@ -215,3 +221,104 @@ export const postOrder = async (
...
+    default:
+      return 'Unknown';
+  }
+}
+
+function formatDate(date) {
+  const formattedDate = new Date(date).toLocaleString();
+
+  return formattedDate;
+}
```
Comment (huongnt-2545) [inline src/controllers/Order.controller.ts]: function để ở folder utils/ dùng chung

File: src/controllers/Order.controller.ts [hunk 3]
```diff
@@ -215,3 +221,104 @@ export const postOrder = async (
...
+
+// GET view history order
+export const getOderList = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const user = await checkLoggedIn(req, res);
+    const page = parseInt(req.query.page as string) || 1;
```
Comment (huongnt-2545) [inline src/controllers/Order.controller.ts]: `1` -> e khai báo trong file constants, ko nên hard code

File: src/routes/order.toute.ts
Comment (huongnt-2545) [inline src/routes/order.toute.ts:1]: file name -> `order.route.ts`

**Comment by hoangnq-3414** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**